pointlio_mapping:
  ros__parameters:
    common:
      lid_topic:  "/livox/lidar" 
      imu_topic:  "/livox/imu" 
      con_frame: false
      con_frame_num: 1
      cut_frame: false
      cut_frame_time_interval: 0.1
      time_diff_lidar_to_imu: 0.0

    preprocess:
      lidar_type: 5  # MID360
      scan_line: 4
      timestamp_unit: 3
      blind: 0.3

    mapping:
      imu_en: true
      extrinsic_est_en: false
      imu_time_inte: 0.01  # 100Hz IMU
      lidar_time_inte: 0.1  # 10Hz LiDAR
      satu_acc: 3.0
      satu_gyro: 35
      acc_norm: 1.0
      lidar_meas_cov: 0.01
      acc_cov_output: 500
      gyr_cov_output: 1000 
      b_acc_cov: 0.0001 
      b_gyr_cov: 0.0001 
      imu_meas_acc_cov: 0.1
      imu_meas_omg_cov: 0.1
      gyr_cov_input: 0.01
      acc_cov_input: 0.1
      plane_thr: 0.1
      match_s: 81
      ivox_grid_resolution: 2.0
      gravity: [0.0, 0.0, -9.810]
      gravity_init: [0.0, 0.0, -9.810]
      # MID360 extrinsic parameters
      extrinsic_T: [ -0.011, -0.02329, 0.04412 ]
      extrinsic_R: [ 1, 0, 0,
                     0, 1, 0,
                     0, 0, 1 ]

    odometry: 
      publish_odometry_without_downsample: false

    publish:
      path_en: true
      scan_publish_en: true
      scan_bodyframe_pub_en: true

    pcd_save:
      pcd_save_en: true
      interval: 5